### PR TITLE
Add plaid link action and controller

### PR DESCRIPTION
**Pull request by vinothvino42**
In this PR, I added the plaid link action

**Sample YAML**
```yaml
Button:
  label: Link Account
  onTap:
    openPlaidLink:
      id: plaidId
      linkToken: YOUR LINK TOKEN
      onSuccess: |
        //@code
        console.log("Account Linked Successfully");
        console.log(event.data);
        console.log("Public Token" + event.data['publicToken']);

      onEvent: |
        //@code
        console.log("All Events");
        console.log(event.data);

      onExit: |
        //@code
        console.log("Something went wrong!");
        console.log(event.data);
```

`event.data` will be different for each callback. 
onSuccess - [linkSuccess object](https://plaid.com/docs/link/ios/#link-ios-onsuccess-linkSuccess)
onEvent - [linkEvent object](https://plaid.com/docs/link/ios/#link-ios-onevent-linkEvent)
onExit - [linkExit object](https://plaid.com/docs/link/ios/#link-ios-onexit-linkExit)

Please check out the doc for more info - https://plaid.com/docs/link/ios/#onsuccess

### CODE REVIEW

File: lib/screen_controller.dart [hunk 1]
```diff
@@ -185,6 +188,51 @@ class ScreenController {
...
+            }
+          },
+          (linkEvent) {
+            if (action.onEvent != null) {
+              executeActionWithScope(
+                context,
+                scopeManager!,
+                action.onEvent!,
+                event: EnsembleEvent(action.initiator!,
+                    data: {'eventData': linkEvent.toJson()}),
```
Comment (vusters) [inline lib/screen_controller.dart:216]: by convention we always do `event.data.*`. So do `data: linkEvent.toJson()`. Same with others

File: lib/widget/plaid_link/plaid_link_controller.dart [hunk 2]
```diff
@@ -0,0 +1,36 @@
+import 'package:flutter/material.dart';
+import 'package:plaid_flutter/plaid_flutter.dart';
+
+typedef PaidLinkSuccessCallback = void Function(LinkSuccess);
```
Comment (vusters) [inline lib/widget/plaid_link/plaid_link_controller.dart:4]: spelling

File: lib/framework/action.dart [hunk 3]
```diff
@@ -211,6 +211,39 @@ class ShowBottomModalAction extends EnsembleAction {
...
+
+  factory PlaidLinkAction.fromYaml({Invokable? initiator, YamlMap? payload}) {
+    if (payload == null || payload['options']['linkToken'] == null) {
+      throw LanguageError(
+          "${ActionType.openPlaidLink.name} requires the plaid's link_token");
+    }
+
+    return PlaidLinkAction(
+      initiator: initiator,
+      options: Utils.getMap(payload['options']),
```
[the remark this answers is not in the document]
Comment (vinothvino42) [reply lib/framework/action.dart:239]: Okay

File: lib/screen_controller.dart (hunk 1 above)
Comment (vinothvino42) [reply lib/screen_controller.dart:216]: We have three callbacks with data, so I thought to split it evenly with their name like eventData, successData, errorData. Okay I'll add every object directly to the data param.

File: lib/widget/plaid_link/plaid_link_controller.dart (hunk 2 above)
Comment (vinothvino42) [reply lib/widget/plaid_link/plaid_link_controller.dart:4]: Okay I'll fix it

**Comment by vusters** [comment]
One more thing. Please create a new Module, call it `ensemble_connect` and put the Plaid code in here. The base ensemble shouldn't have any reference to Plaid. See existing modules like 'ensemble_auth' for example.

File: lib/screen_controller.dart (hunk 1 above)
Comment (vinothvino42) [reply lib/screen_controller.dart:216]: Done

File: lib/widget/plaid_link/plaid_link_controller.dart (hunk 2 above)
Comment (vinothvino42) [reply lib/widget/plaid_link/plaid_link_controller.dart:4]: Done

**Comment by vinothvino42** [comment]
@vusters @kmahmood74 Please see the above updated YAML EDL. 
@vusters I updated the changes which you've mentioned here as a comment and I just moved the paid link service to a separate package - https://github.com/EnsembleUI/ensemble_connect

You can enable/disable the paid link service as mentioned here - https://github.com/EnsembleUI/starter#modules
```dart
if (usePlaidLink) {
      // Uncomment to enable plaid link service
      GetIt.I.registerSingleton<PlaidLinkManager>(PlaidLinkManagerImpl());
    } else {
      GetIt.I.registerSingleton<PlaidLinkManager>(PlaidLinkManagerStub());
    }
```

**Comment by vusters** [comment]
Ensemble_Live requires all module to turn on. Please enable Connect too - see existing example.

**Comment by vinothvino42** [comment]
> Ensemble_Live requires all module to turn on. Please enable Connect too - see existing example.

I'll add that once this PR merged it. Because in the preview it supposed to map to main branch, this one is in plaid-integration branch

**Review by vusters — APPROVED** [review]

**Pull request merged**